File: feature/tournaments/src/main/java/n7/ad2/tournaments/internal/db/GamesDao.java
//package n7.ad2.tournaments.internal.db;
//
//import androidx.lifecycle.LiveData;
//import androidx.paging.DataSource;
//import androidx.room.Dao;
//import androidx.room.Insert;
//import androidx.room.OnConflictStrategy;
//import androidx.room.Query;
//
//import java.util.List;
//
////@Dao
////public interface GamesDao {
////
////    @Insert(onConflict = OnConflictStrategy.IGNORE)
////    long[] setGames(List<TournamentGame> object);
////
////    @Query("DELETE FROM TournamentGame")
////    void deleteAll();
////
////    @Query("DELETE FROM TournamentGame where teamTimeRemains!=0")
////    void deleteAllUnfinished();
////
////    @Query("SELECT * FROM TournamentGame where url=:url")
////    LiveData<TournamentGame> getGameByUrl(String url);
////
////    @Query("SELECT * FROM TournamentGame")
////    DataSource.Factory<Integer, TournamentGame> getDataSourceGames();
////
////}

File: feature/tournaments/src/main/kotlin/n7/ad2/tournaments/internal/db/GamesDao.kt
package n7.ad2.tournaments.internal.db

import androidx.lifecycle.LiveData
import androidx.paging.DataSource
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface GamesDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun setGames(games: List<TournamentGame>): LongArray

    @Query("DELETE FROM TournamentGame")
    fun deleteAll()

    @Query("DELETE FROM TournamentGame where teamTimeRemains!=0")
    fun deleteAllUnfinished()

    @Query("SELECT * FROM TournamentGame where url=:url")
    fun getGameByUrl(url: String): LiveData<TournamentGame>

    @Query("SELECT * FROM TournamentGame")
    fun getDataSourceGames(): DataSource.Factory<Int, TournamentGame>
}
